verifier/verifier.go

```diff
@@ -518,6 +545,67 @@ func verifyAuthenticTimestamp(outcome *notation.VerificationOutcome) *notation.V
 	}
 }
 
+func verifyRevocation(outcome *notation.VerificationOutcome, client *http.Client, logger log.Logger) *notation.ValidationResult {
+	r, err := revocation.New(client)
+	if err != nil {
+		logger.Debug("no HTTP client provided for revocation checking")
+		return &notation.ValidationResult{
+			Type:   trustpolicy.TypeRevocation,
+			Action: outcome.VerificationLevel.Enforcement[trustpolicy.TypeRevocation],
+			Error:  errors.New("unable to check revocation status, an HTTP client must be provided"),
+		}
+	}
+
+	authenticSigningTime := outcome.EnvelopeContent.SignerInfo.SignedAttributes.SigningTime
+	// TODO use authenticSigningTime from signerInfo
+	// https://github.com/notaryproject/notation-core-go/issues/38
+	revocationResult := r.Validate(outcome.EnvelopeContent.SignerInfo.CertificateChain, authenticSigningTime)
+
+	result := &notation.ValidationResult{
+		Type:   trustpolicy.TypeRevocation,
+		Action: outcome.VerificationLevel.Enforcement[trustpolicy.TypeRevocation],
+	}
+
+	// Log all non-nil errors as debug
+	for i, certResult := range revocationResult {
+		for _, err := range certResult.ServerResults {
+			if err != nil {
+				logger.Debugf("error for certificate #%d in chain: %v", (i + 1), err)
+			}
+		}
+	}
+
+	currResult := revocation.OK
+	for _, certResult := range revocationResult {
+		switch certResult.Result {
+		case revocation.OK:
+			continue
+		case revocation.Revoked:
+			currResult = revocation.Revoked
+			// will break out of loop after this
+		default:
+			// revocation.Unknown
+			currResult = revocation.Unknown
+			continue
+		}
+		if currResult == revocation.Revoked {
+			break
+		}
+	}
+
+	switch currResult {
+	case revocation.OK:
+		logger.Debug("no important errors encountered while checking revocation, status is OK")
+	case revocation.Revoked:
+		result.Error = errors.New("one or more certificates in the chain were determined to be revoked, status is Revoked")
```

Comment:
```suggestion
		result.Error = errors.New("signing certificate(s) revoked")
```

Reply:
Can we include the certificate subject that was revoked. Otherwise, users will need to parse the signatures to debug issues.

Reply:
The subject can be fairly long for an error message. I think that this may cause difficulties for understanding what went wrong (especially if there are multiple certificates. I believe a better alternative would be to add the subject of the cert to the debug log for each error instead in addition to the number of the cert in the chain.

Reply:
I don't think the error messages with subject names would be of unreasonable length. Notation already includes certificate subject names in various error messages.